frontend/packages/knative-plugin/src/components/add/eventSource-validation-utils.ts

```diff
@@ -71,10 +71,17 @@ export const sourceDataSpecSchema = yup
     }),
   });
 
-export const eventSourceValidationSchema = yup.object().shape({
-  project: projectNameValidationSchema,
-  application: applicationNameValidationSchema,
-  name: nameValidationSchema,
-  sink: sinkServiceSchema,
-  data: sourceDataSpecSchema,
+export const eventSourceValidationSchema = yup.lazy((formData) => {
+  if (Object.keys(EventSources).includes(formData.type)) {
```

Comment:
why not use util which you added `isKnownEventSource` ? and any benefit of `yup.lazy((formData)` cc @rohitkrai03

Reply:
using lazy gave access to the `values` that could be used
used `isKnownEventSource` for check in the if block

Reply:
@nemesis09 i don't see it updated